ledger/trie/iterator.go

```diff
@@ -97,8 +97,13 @@ func (i *NodeIterator) Next() bool {
 		// As we descend into the left child with priority, the only case where we still dig into the
 		// right child is if `n` is `p`'s left child.
 		parent := i.peek()
-		if parent.LeftChild() == n {
-			i.dig(parent.RightChild())
+		switch p := parent.(type) {
+		case *Branch:
+			if p.left == n {
+				i.dig(p.right)
+			}
+		case *Extension, *Leaf:
+			// do nothing
```

Comment:
Got to change everything that used `RightChild()` and `LeftChild()` since they got removed from the interface.